pkg/eval/vals/num.go

```diff
@@ -201,13 +197,11 @@ func ConvertToFloat64(num Num) float64 {
 	case int:
 		return float64(num)
 	case *big.Int:
-		if num.IsInt64() {
-			// Might fit in float64
+		if num.IsInt64() { // might fit in float64
+			// TODO: Make this more robust so the "might fit" is "will fit".
```

Comment:
There is no loss of robustness here. If the number overflows `float64`, the conversion below will result in +Inf or -Inf. I guess the comment isn't clear about that, I'll expand the comment.